prow/plugins/trigger/pull-request.go

```diff
@@ -141,10 +149,63 @@ func handlePR(c Client, trigger plugins.Trigger, pr github.PullRequestEvent) err
 			}
 			return buildAll(c, &pr.PullRequest, pr.GUID, *trigger.ElideSkippedContexts, baseSHA, presubmits)
 		}
+	case github.PullRequestActionClosed:
+		if err := abortAllJobs(c, &pr.PullRequest); err != nil {
+			c.Logger.WithError(err).Error("Failed to abort jobs for closed pull request")
+			return err
+		}
 	}
 	return nil
 }
 
+func abortAllJobs(c Client, pr *github.PullRequest) error {
+	selector, err := labelSelectorForPR(pr)
+	if err != nil {
+		return fmt.Errorf("failed to construct label selector: %w", err)
+	}
+
+	listOpts := metav1.ListOptions{LabelSelector: selector.String()}
+	jobs, err := c.ProwJobClient.List(listOpts)
+	if err != nil {
+		return fmt.Errorf("failed to list prowjobs for pr: %w", err)
+	}
+
+	var errs []error
+	for _, job := range jobs.Items {
+		// Do not abort jobs that already completed
+		if job.Complete() {
+			continue
+		}
+		job.Status.State = prowapi.AbortedState
+		if _, err := c.ProwJobClient.Update(&job); err != nil {
```

Comment:
I've explicitly chosen to `Update` and not to `Patch` so this will fail if another actor changed the job in the meantime. Consequently, this is a best-effort solution that may fail incorrectly in some cases like `crier` updating the job wrong. IMHO that is better allowing races with the responsible `agent` for the jobs status.

Reply:
Can we add this comment to the code?